app/src/main/java/com/example/diceroller/DiceFragment.kt

```diff
@@ -1,82 +1,73 @@
 package com.example.diceroller
 
 import android.os.Bundle
-import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.TextView
 import androidx.core.content.ContextCompat
-import androidx.core.graphics.drawable.toDrawable
+import androidx.fragment.app.Fragment
 import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.lifecycleScope
-import androidx.lifecycle.whenCreated
+import androidx.navigation.fragment.findNavController
 import com.example.diceroller.databinding.FragmentDiceBinding
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.collectLatest
-import kotlinx.coroutines.flow.observeOn
+import kotlinx.coroutines.flow.collect
 
 class DiceFragment : Fragment() {
-    var diceType:MutableStateFlow<DiceType> = MutableStateFlow(DiceType.D6)
-    private var _binding: FragmentDiceBinding? = null
-    private val binding get() = _binding!!
-
-    private lateinit var viewModel: RolledDiceViewModel
+    private lateinit var binding: FragmentDiceBinding
+    private lateinit var viewModel: DiceRollerViewModel
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        viewModel = ViewModelProvider(requireActivity())[DiceRollerViewModel::class.java]
+
+        observeState()
     }
 
     override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
+        inflater: LayoutInflater,
+        container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View {
-        // Inflate the layout for this fragment
-        _binding = FragmentDiceBinding.inflate(inflater, container, false)
-        val view = binding.root
-
-        binding.rollButton.setOnClickListener {
-            recieveDice()
-        }
-        return view
+    ): View = FragmentDiceBinding.inflate(inflater, container, false).run {
+        binding = this
+        this.root
     }
 
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
```

Comment:
Is there a specific reason you decided to call it inside the `onCreate()` function instead of inside `onViewCreated()`? or is it a bad idea to do so? I'm a little bit confused on what's the best practices are in regards to this are